include/no_os_i3c.h

```diff
@@ -0,0 +1,446 @@
+#define NO_OS_I3C_CCC_GETDCR		(NO_OS_I3C_CCC_ID(0xf, NO_OS_I3C_CCC_DIRECT) | NO_OS_I3C_CCC_SET_LEN(1) | NO_OS_I3C_CCC_SET_RNW(1))
+#define NO_OS_I3C_CCC_GETSTATUS		(NO_OS_I3C_CCC_ID(0x10, NO_OS_I3C_CCC_DIRECT) | NO_OS_I3C_CCC_SET_LEN(2) | NO_OS_I3C_CCC_SET_RNW(1))
+#define NO_OS_I3C_CCC_GETACCMST		NO_OS_I3C_CCC_ID(0x11, NO_OS_I3C_CCC_DIRECT)
+#define NO_OS_I3C_CCC_SETBRGTGT		NO_OS_I3C_CCC_ID(0x13, NO_OS_I3C_CCC_DIRECT)
+#define NO_OS_I3C_CCC_GETMXDS		NO_OS_I3C_CCC_ID(0x14, NO_OS_I3C_CCC_DIRECT)
+#define NO_OS_I3C_CCC_GETHDRCAP		NO_OS_I3C_CCC_ID(0x15, NO_OS_I3C_CCC_DIRECT)
+#define NO_OS_I3C_CCC_GETXTIME		NO_OS_I3C_CCC_ID(0x19, NO_OS_I3C_CCC_DIRECT)
+
+#define NO_OS_I3C_CCC_EVENT_SIR		NO_OS_BIT(0)
+#define NO_OS_I3C_CCC_EVENT_MR		NO_OS_BIT(1)
+#define NO_OS_I3C_CCC_EVENT_HJ		NO_OS_BIT(3)
+/**
+ * List of possible IRQ events.
+ */
+#define NO_OS_I3C_IRQ_IBI   0x1
+#define NO_OS_I3C_IRQ_HJ    0x2
+#define NO_OS_I3C_IRQ_CR    0x4
+
+/**
+ * @enum no_os_i3c_slot_status - I3C address slot status
+ *
+ * @brief On an I3C bus, addresses are assigned dynamically, and we need to
+ * know which addresses are free to use and which ones are already assigned.
+ *
+ * Addresses marked as reserved are those reserved by the I3C protocol
+ * (broadcast address, ...).
+ */
+enum no_os_i3c_slot_status {
+	/** Address is free */
+	NO_OS_I3C_ADDR_SLOT_FREE,
+	/** Address is reserved */
+	NO_OS_I3C_ADDR_SLOT_RSVD,
+	/** Address is assigned to an I2C device */
+	NO_OS_I3C_ADDR_SLOT_I2C_DEV,
+	/** Address is assigned to an I3C device */
+	NO_OS_I3C_ADDR_SLOT_I3C_DEV,
+	/** Address slot mask */
+	NO_OS_I3C_ADDR_SLOT_STATUS_MASK = 3,
+};
+
+/**
+ * @struct no_os_i3c_bcr_flags
+ */
+struct no_os_i3c_bcr_flags {
+	uint8_t MAX_DATA_SPEED_LIMITATION	: 1;
+	uint8_t IBI_REQUEST_CAPABLE		: 1;
+	uint8_t IBI_PAYLOAD			: 1;
+	uint8_t OFFLINE_CAPABLE			: 1;
+	uint8_t VIRTUAL_TARGET_SUPPORT		: 1;
+	uint8_t ADVANCED_CAPABILITIES		: 1;
+	uint8_t DEVICE_ROLE			: 2;
+};
+
+/**
+ * @union no_os_i3c_bcr
+ * @brief Stores a BCR.
+ */
+union no_os_i3c_bcr {
+	struct no_os_i3c_bcr_flags fields;
+	uint8_t value;
+};
+
+/**
+ * @struct no_os_i3c_irq_flags
+ */
+struct no_os_i3c_irq_flags {
+	uint8_t IBI	: 1;
+	uint8_t HJ	: 1;
+	uint8_t CR	: 1;
+};
+
+/**
+ * @union no_os_i3c_irq_events
+ * @brief Stores an IRQ event.
+ */
+union no_os_i3c_irq_events {
+	struct no_os_i3c_irq_flags fields;
+	uint8_t value;
+};
+
+/**
+ * @union no_os_i3c_ccc_info
+ */
+struct no_os_i3c_ccc_info {
+	/** I3C bus receive Target Address during IBI or Bus Role Request event. */
+	uint32_t ibi_cr_addr;
+	/** I3C bus get Number of Data Payload after an IBI event. */
+	uint32_t ibi_payload_len;
+	/** I3C bus received IBI Payload. */
+	uint32_t ibi_payload;
+};
+
+/**
+ * @struct no_os_i3c_daa_lut
+ * @brief Stores the PID + DA information to look-up during the DAA.
+ * Should be used only for device initialization, since it may
+ * go outdated.
+ */
+struct no_os_i3c_daa_lut {
```

Comment:
I still see this one containing bit fields and i think they should also be remove

Besides the endianess issue there is also a risk when serializing this info and sending it to another device (like a PC) that runs software to interpret the info. The PC software could be compiled with different compiler than the firmware and the packing, order, padding etc. of the structures with bitfields may differ.